.github/workflows/ci.yml

```diff
@@ -16,8 +17,19 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  authorize:
+    environment:
+      ${{ github.event_name == 'pull_request_target' &&
+      github.event.pull_request.head.repo.full_name != github.repository &&
+      matrix.os == 'macos-latest' &&
+      'external' || 'internal' }}
```

Comment:
@mhils external environment requires approval, I don't know if this is safe enough, and I don't know if it would be a viable solution, but it is an idea

Reply:
This kind of stuff is hard to get secure, and would also mean we need to manually approve all pull requests. Let's not do that. 😅 

Can we just not sign the binary when the keys are not present? Or use some mock/test keys? I realize this means the produced binaries won't work, but that's fine. It still asserts that they compile cleanly, and as long as we have working binaries from the main branch that serves our purpose. :)